airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source.py

```diff
@@ -142,7 +138,7 @@ def _handle_item(
         concurrent_stream_processor: ConcurrentReadProcessor,
     ) -> Iterable[AirbyteMessage]:
         # handle queue item and call the appropriate handler depending on the type of the queue item
-        if isinstance(queue_item, Exception):
+        if isinstance(queue_item, StreamThreadException):
```

Comment:
Python/dynamically type language can be so complicated just because of typing...

The issue with catching `Exception` is that we won't know from which thread it comes from. What would you propose in terms of how we should handle generic exceptions?

Reply:
I see the same problem you see. it wouldn't be easy to resolve, but at least we'd know we made a mistake by logging it and/or failing loudly

Reply:
Makes sense. Do you have a way to fail loudly in that case? I can break the main thread with a system_error but this could lead to stuck syncs again and wouldn't be raised by the platform as they expect return code != 0